File: Dev/C++/Mascaret/src/VEHA/Behavior/Time/TimeExpression.cpp
#include "VEHA/Behavior/Time/TimeExpression.h"

#ifdef _WIN32
#include "Tools/utils.h"
#else
#include <sys/select.h>
#endif
#include <sys/time.h>

#include <string>
#include <sstream>

using std::string;
using std::stringstream;

namespace VEHA
{

TimeExpression::TimeExpression():
    _time(0.0)
{
}

TimeExpression::TimeExpression(double time):
    _time(time)
{
}

TimeExpression::TimeExpression(int year, int month, int day, int hour, int min, int sec)
{
    struct tm date;
    date.tm_year = year-1900;
    date.tm_mon = month-1;
    date.tm_mday = day;
    date.tm_hour = hour;
    date.tm_min = min;
    date.tm_sec = sec;
    date.tm_isdst = -1;
    _time = (double)mktime(&date);
}

TimeExpression::~TimeExpression()
{
}

double TimeExpression::getAsDouble() const
{
    return _time;
}

string TimeExpression::getAsString() const
{
    stringstream stream;
    time_t tmp = (time_t)_time;
    stream << ctime(&tmp);
    return stream.str().erase(stream.str().size()-1);
}

TimeExpression TimeExpression::createTimeOfDay()
{
    struct timeval tv;
    struct timezone tz;
    gettimeofday(&tv,&tz);
    return TimeExpression(((double)tv.tv_sec)+(((double)tv.tv_usec)*1e-6));
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/Time/TimeExpression.cs
using System;

namespace Mascaret
{
    public class TimeExpression
    {

        private long timeExp;
        public long TimeExp
        {
            get { return timeExp; }
            set { timeExp = value; }
        }

        public TimeExpression(int year, int month, int day, int hour, int min, int sec)
        {
            DateTime time = new DateTime(year, month, day, hour, min, sec);
            if (time != null) timeExp = time.Ticks / 10000;
        }

        /*public void instantiate(double dt)
        {
		
        }
		
        public void setValue(double time)
        {
			
        }
	
        public string getAsString()
        {
            return timeExp.ToString();
        }*/

        public static TimeExpression createTimeOfDay()
        {
            TimeExpression time = new TimeExpression(2014, 5, 1, 0, 0, 0);
            //time.TimeExp = DateTime.Now.ToOADate();
            time.TimeExp = DateTime.Now.Ticks / 10000;
            return time;
        }
    }
}


